src/main/java/org/springframework/data/couchbase/core/WithExpiry.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.couchbase.core;
+
+import java.time.Duration;
+
+/**
+ * A common interface for all of Insert, Replace, Upsert that take expiry
+ *
+ * @author Michael Reiche
+ *
+ * @param <T> - the entity class
+ */
+public interface WithExpiry<T> {
+  Object withExpiry(Duration expiry);
```

Comment:
How would you express that a default expiry should be overridden to persistent (i.e. no expiry)?